### PR TITLE
Fix few clippy warnings

### PR DESCRIPTION
**Pull request by overvenus**
PTAL @siddontang @disksing

### CODE REVIEW

**Comment by siddontang** [comment]
CI failed

**Comment by overvenus** [comment]
It looks like clippy is outdated, I should wait few days. 🤷

**Comment by siddontang** [comment]
CI still failed.

**Comment by siddontang** [comment]
why not using a special nightly and clippy version like TiKV does?

**Comment by overvenus** [comment]
> If you’re building a library that other projects will depend on, put Cargo.lock in your .gitignore. 

http://doc.crates.io/guide.html#cargotoml-vs-cargolock

TiKV specifies clippy by Cargo.lock. rust-prometheus as a library should not contain a Cargo.lock in the repo.

File: src/push.rs [hunk 1]
```diff
@@ -80,6 +80,10 @@ fn push(job: &str,
         method: &str)
         -> Result<()> {
 
+    // Suppress clippy warning needless_pass_by_value.
+    let grouping = grouping;
+    let mfs = mfs;
```
Comment (disksing) [inline src/push.rs]: Why not pass by reference instead?

Comment (overvenus) [inline src/push.rs]: Then other functions will cause `needless_pass_by_value`.

https://github.com/pingcap/rust-prometheus/pull/99/files/7e0264b9acbca0ff2ceb719aec04f84e00779c37#diff-895850e1c4456e2e576a20e2f83bc96aR70

Comment (disksing) [inline src/push.rs]: Can we use `AsRef` to accept both value and reference? https://doc.rust-lang.org/std/convert/trait.AsRef.html

Comment (siddontang) [inline src/push.rs]: 🤔  I prefer simple here.

**Comment by siddontang** [comment]
LGTM

File: src/push.rs (hunk 1 above)
Comment (disksing) [inline src/push.rs]: IMO, as a library, it should minimize its overhead, so when it does not need ownership, it should not require user to provide.

Comment (overvenus) [inline src/push.rs]: Sure, but taking a reference will change APIs. To fix it, I am going to send a PR separately.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**